src/d/actor/d_a_player_particle.inc

```diff
@@ -106,22 +136,147 @@ void daPy_swimTailEcallBack_c::draw(JPABaseEmitter*) {
+    GXLoadTexObj(&texObj, GX_TEXMAP0);
+    GXLoadPosMtxImm(mMtx, GX_PNMTX0);
+    GXSetCurrentMtx(GX_PNMTX0);
+    GXSetTevOrder(GX_TEVSTAGE0, GX_TEXCOORD0, GX_TEXMAP0, GX_COLOR_NULL);
+    GXCallDisplayList(&l_sightDL, sizeof(l_sightDL) - 0x09);
+    J3DShape::resetVcdVatCache();
 }
 
 /* 8012952C-801296FC       .text setSight__18daPy_sightPacket_cFv */
 void daPy_sightPacket_c::setSight() {
-    /* Nonmatching */
+    cXyz proj;
+    mDoLib_project(&mPos, &proj);
+    mDoMtx_stack_c::transS(proj);
+
+    if (mLockFlag) {
+        f32 var_f2;
+        f32 var_f31;
+        if (field_0x6 < 13) {
+            var_f2 = 1.0f - ((s16)field_0x6 / 13.0f);
+            var_f31 = -var_f2;
+        } else {
+            var_f2 = (field_0x6 - 13) / 13.0f;
+            var_f31 = var_f2;
+        }
+
+        mLockAlpha = 150.0f + (30.0f * var_f2);
+        f32 temp_f3 = 0.65f + (0.35f * var_f2);
+
+        mDoMtx_stack_c::scaleM(temp_f3 * mpImg->width, temp_f3 * mpImg->height, temp_f3);
+        mDoMtx_stack_c::ZrotM(32768.0f * cM_ssin(16384.0f * var_f31));
+    } else {
+        mDoMtx_stack_c::scaleM(10.0f, 10.0f, 10.0f);
+    }
+
+    mDoMtx_copy(mDoMtx_stack_c::get(), mMtx);
+    dComIfGd_set2DXlu(this);
 }
 
 /* 801296FC-801297F4       .text initSwBlur__13daPy_swBlur_cFPA4_fifi */
-void daPy_swBlur_c::initSwBlur(MtxP, int, f32, int) {
-    /* Nonmatching */
+void daPy_swBlur_c::initSwBlur(MtxP mtx, int idx, f32 blurTopRate, int blurColorType) {
+    int i = idx * 2;
+
+    mBlurTopRate = blurTopRate;
+    mBlurColorType = blurColorType;
+
+    mDoMtx_multVec(mtx, mpPosBuffer + i, &field_0x034[0]);
+    mDoMtx_multVec(mtx, mpPosBuffer + i + 1, &field_0x304[0]);
+
+    field_0x034[0] += (field_0x034[0] - field_0x304[0]) * mBlurTopRate;
+    field_0x034[1] = field_0x034[0];
+    field_0x304[1] = field_0x304[0];
+    field_0x014 = 0;
+    field_0x018 = idx;
+    field_0x028.set(mtx[0][3], mtx[1][3], mtx[2][3]);
 }
 
 /* 801297F4-80129A78       .text copySwBlur__13daPy_swBlur_cFPA4_fi */
-void daPy_swBlur_c::copySwBlur(MtxP, int) {
-    /* Nonmatching */
+void daPy_swBlur_c::copySwBlur(MtxP mtx, int param_2) {
+    int var_r31 = param_2 - field_0x018;
+    field_0x018 = param_2;
+
+    int var_r30;
+    if (var_r31 > 0) {
+        var_r30 = -2;
+
+        if (var_r31 > 60) {
+            var_r31 = 60;
+        }
+    } else if (var_r31 < 0) {
+        var_r30 = 2;
+
+        if (var_r31 < -60) {
+            var_r31 = 60;
+        } else {
+            var_r31 = -var_r31;
+        }
+    } else {
+        var_r31 = 10;
+        var_r30 = 0;
+    }
+
+    for (int i = 59 - var_r31; i >= 0; i--) {
+        field_0x034[i + var_r31] = field_0x034[i];
+        field_0x304[i + var_r31] = field_0x304[i];
+    }
+
+    // Probably fakematch
+    f32 var_f31 = 0.0f;
+    f32 frac = 1.0f / var_r31;
+    int i;
+    int buffIdx = field_0x018 * 2;
+    cXyz sp50(mtx[0][3], mtx[1][3], mtx[2][3]);
```

Comment:
also might be
```suggestion
    cXyz sp50;
    mDoMtx_multVecZero(mtx, &sp50);
```
etc, there's around 10 other instances in this file that might be this too